src/System.AppContext/src/System/AppContext.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using Windows.ApplicationModel;
+
+namespace System
+{
+    public static class AppContext
+    {
+        [Flags]
+        private enum SwitchValueState
+        {
+            HasFalseValue = 0x1,
+            HasTrueValue = 0x2,
+            HasLookedForOverride = 0x4, // Not used on .NET Native
+            UnknownValue = 0x8 // Not used on .NET Native
+        }
+
+        private static Dictionary<string, SwitchValueState> s_switchMap = new Dictionary<string, SwitchValueState>();
```

Comment:
Ok.   Though regarding "The overhead of the ConcurrentDictionary was too great for the use case", I question that.  It has higher memory overheads, sure, but reading from it (which I expect would be the most common case) should be measurably faster than taking a lock to access a Dictionary.